File: blockball-bukkit-api/src/main/java/com/github/shynixn/blockball/api/bukkit/event/BallDeathEvent.java
package com.github.shynixn.blockball.api.bukkit.event;

import com.github.shynixn.blockball.api.entities.Ball;
import com.github.shynixn.blockball.lib.SEvent;

public final class BallDeathEvent extends SEvent {
	private final Ball ball;

	public BallDeathEvent(Ball ball) {
		super();
		this.ball = ball;
	}

	public Ball getBall() {
		return this.ball;
	}
}

File: blockball-bukkit-api/src/main/java/com/github/shynixn/blockball/api/bukkit/event/BallDeathEvent.kt
package com.github.shynixn.blockball.api.bukkit.event

import com.github.shynixn.blockball.api.business.proxy.BallProxy

/**
 * Event which gets sent when the ball is requested to get removed.
 */
class BallDeathEvent(ball: BallProxy) : BallEvent(ball)
